modules/images/dominant-color/load.php

```diff
@@ -0,0 +1,368 @@
+		if ( count( $attachment_ids ) > 1 ) {
+			/*
+			 * Warm the object cache with post and meta information for all found.
+			 * images to avoid making individual database calls.
+			 */
+			_prime_post_caches( $attachment_ids, false, true );
+		}
+
+		// Iterate through the matches in order of occurrence as it is relevant for whether or not to lazy-load.
+		$filtered_image = null;
+		foreach ( $matches as $match ) {
+			// Filter an image match.
+			if ( empty( $images[ $match[0] ] ) ) {
+				continue;
+			}
+
+			$filtered_image = $match[0];
+			$attachment_id  = $images[ $match[0] ];
+			$filtered_image = dominant_color_img_tag_add_dominant_color( $filtered_image, $context, $attachment_id );
+
+			if ( null !== $filtered_image && $filtered_image !== $match[0] ) {
+				$content = str_replace( $match[0], $filtered_image, $content );
+			}
+		}
+		return $content;
+	}
+
+	$filters = array( 'the_content', 'the_excerpt', 'widget_text_content', 'widget_block_content' );
+	foreach ( $filters as $filter ) {
+		add_filter( $filter, 'dominant_color_filter_content_tags', 20 );
+	}
+}
+
+/**
+ * Add CSS needed for to show the dominant color as an image background.
+ *
+ * @since n.e.x.t
+ */
+function dominant_color_add_inline_style() {
+	$handle = 'dominant-color-styles';
+	wp_register_style( $handle, false );
+	wp_enqueue_style( $handle );
+	$custom_css = 'img[data-dominantcolor]:not(.has-transparency) { background-color: var(--dominant-color); background-clip: content-box, padding-box; }';
+	wp_add_inline_style( $handle, $custom_css );
+}
+add_filter( 'wp_enqueue_scripts', 'dominant_color_add_inline_style' );
+
+
+/**
+ * Overloads wp_image_editors() to load the extended classes.
+ *
+ * @since n.e.x.t
+ *
+ * @return string[] Registered image editors class names.
+ */
+function dominant_color_set_image_editors() {
+        if( ! class_exists( 'Dominant_Color_Image_Editor_GD' ) ) {
+	    require_once 'class-dominant-color-image-editor-gd.php';
+	}
+        if( ! class_exists( 'Dominant_Color_Image_Editor_Imagick' ) ) {
+	    require_once 'class-dominant-color-image-editor-imagick.php';
+	}
+
+	return array( 'Dominant_Color_Image_Editor_GD', 'Dominant_Color_Image_Editor_Imagick' );
+}
+
+/**
+ * Get dominant color of image
+ *
+ * @since n.e.x.t
+ *
+ * @param integer $attachment_id the image id.
+ * @return string|null the dominant color of the image. or null if no color is found.
+ */
+function dominant_color_get_dominant_color( $attachment_id ) {
+
+	add_filter( 'wp_image_editors', 'dominant_color_set_image_editors' );
+
+	$file   = get_attached_file( $attachment_id );
+	$editor = wp_get_image_editor( $file );
+
+	if ( ! is_wp_error( $editor ) && method_exists( $editor, 'dominant_color_get_dominant_color' ) ) {
+		$dominant_color = $editor->dominant_color_get_dominant_color();
+		if ( ! is_wp_error( $dominant_color ) ) {
+			return $dominant_color;
+		}
+	}
+
+	remove_filter( 'wp_image_editors', 'dominant_color_set_image_editors' );
+
+	return null;
```

Comment:
```suggestion
	if ( is_wp_error( $editor ) || method_exists( $editor, 'dominant_color_get_dominant_color' ) ) {
		return null;
	}
	
	$dominant_color = $editor->dominant_color_get_dominant_color();
         if ( ! is_wp_error( $dominant_color ) ) {
		return null;
	}

	remove_filter( 'wp_image_editors', 'dominant_color_set_image_editors' );

	return $dominant_color;
```

This is a little easier to read for me.